src/backend/recycledRevoker/blackList.ts

```diff
@@ -0,0 +1,33 @@
+import { SubmittableExtrinsic } from '@kiltprotocol/sdk-js';
+
+import { logger } from '../utilities/logger';
+
+import { generateTransactions } from './generateTransactions';
+
+/**
+ * List of `SubmittableExtrinsic`s to be gradually signed and submitted on batches.
+ *
+ * `SubmittableExtrinsic` are **transactions**, in this case, they could either be **Revocations** or **Removals**.
+ */
+export const blackList: SubmittableExtrinsic[] = [];
+
+export async function fillBlackList(fromBlock: number) {
+  const transactionGenerator = generateTransactions(fromBlock);
+
+  for await (const extrinsicToSubmit of transactionGenerator) {
+    blackList.push(extrinsicToSubmit);
+  }
+}
+
+export async function removeFromBlackList(
+  successfulTransactions: SubmittableExtrinsic[],
+) {
```

Comment:
is just to know what to use as an argument.

Reply:
But it would also work if they are unsuccessful.

Reply:
I think this case is not different from the majority of others, where the lower-level code should not care about its applications, so that it won’t have to be updated in case the logic changes in the other place.